src/lib.rs

```diff
@@ -92,6 +92,7 @@
 //! [examples]: https://github.com/stm32-rs/stm32f1xx-hal/tree/master/examples
 
 #![no_std]
+#![allow(non_camel_case_types)]
```

Comment:
Why do we need this? it seems like sticking to standard rust formatting would be desireable

Reply:
I have been using the same AdcSampleTime enum as in stm32f0xx-hal. So this non_camel_case_types is needed to make rust happy with T_n names. I could drop it and use other names for AdcSampleTime enum, e.g. Tn. But it looks like using non-camel case types is quite common for stm32fxxx HAL implementations: see both stm32f0xx and stm32f4xx.

Reply:
I think I would prefer using `Tn`, but I'd be fine with `T_n`, especially if it is done that way in more crates. However, in that case I would prefer to have the `[allow]` directive on the type rather than as a global thing.